### PR TITLE
Implementation of Rastertiles, removal of GeoServer

### PR DESCRIPTION
**Pull request by mssond**
**Draft 1 Dated 2024-06-06**
### Summary:

**Enhancement:**
1: Auto generate raster tiles and store in dedicated S3 bucket - completed
2: Point Dash at S3 raster tile endpoint instead of geoserver - completed

All reference to GeoServer has been removed
Further cleaned up code
TODO/Remaining Task: add a main function and further cleanup/organize code

**AWS VRRC DEV**
- Created new S3 Bucket called "vrrc-insar-tiles-store-dev"
- Opened up Permission "Block public and cross-account access to buckets and objects through any public bucket or access point policies"
-  **IMPORTANT (REVIEW THIS)** - Added Bucket Policy allowing any items within Resource": "arn:aws:s3:::vrrc-insar-tiles-store-dev/*" to be requested. Potentially make it public for certain IAM users only.
- Added few example tiles in this S3 bucket (Edgecumbe 3M36D and Meager 5M3)
- Site-Dropdown menu was arranged in alphabetical order

**CONFIG FILE CONTENT**
[AWS]
bucketName = vrrc-insar-store
rawBucket =
insarProcessQueue = https://sqs.ca-central-1.amazonaws.com/069989789993/vrrc-insar-q
tiles = https://vrrc-insar-tiles-store-dev.s3.ca-central-1.amazonaws.com

[API]
vrrc_api_ip = localhost:8000

### **METHOD TO GET STARTED and TEST THIS PR**

cd ${DIR}
git clone -b rastertiles_2 https://github.com/Volcano-Risk-Reduction-in-Canada/Volcanic-Interpretation-Workbench

cd ${DIR}/Volcanic-Interpretation-Workbench
clear
ls

cd ${DIR}/Volcanic-Interpretation-Workbench
python3 -m venv venv
sudo apt install python3.10-venv

python3 -m venv venv
source venv/bin/activate
pip3 install -r requirements.txt

cat <<EOF > ${DIR}/Volcanic-Interpretation-Workbench/config.ini
[AWS]
bucketName = vrrc-insar-store
rawBucket =
insarProcessQueue = https://sqs.ca-central-1.amazonaws.com/069989789993/vrrc-insar-q
tiles = https://vrrc-insar-tiles-store-dev.s3.ca-central-1.amazonaws.com

[API]
vrrc_api_ip = localhost:8000
EOF
cat ${DIR}/Volcanic-Interpretation-Workbench/config.ini

cd ${DIR}/Volcanic-Interpretation-Workbench
python scripts/get_latest_coh_matrices.py 
python scripts/get_latest_baselines.py

Open a new terminal and start an aws ssm session of the VRRC API
Open a second terminal and export AWS VRRC DEV ENV variables

Then in second terminal, input the following:
cd ${DIR}/Volcanic-Interpretation-Workbench
source venv/bin/activate
python3 app/dash_app.py

### CODE REVIEW

File: app/Data/Garibaldi/3M23/CoherenceMatrix.csv
Comment (drotheram) [inline app/Data/Garibaldi/3M23/CoherenceMatrix.csv:1]: These files shouldn't be committed as they are living files and should eventually be deprecated by requests to the API

File: app/pages/site.py [hunk 1]
```diff
@@ -459,39 +461,36 @@ def calc_polygon_centroid(coordinates):
...
     first = second - delta
-
     first_str = first.strftime('%Y%m%d')
     second_str = second.strftime('%Y%m%d')
-    layer = f'cite:{first_str}_HH_{second_str}_HH.adf.wrp.geo'
+    layer = (
+        f'{TILES_BUCKET}/{SITE}/{BEAM}/{first_str}_{second_str}/'
+        '{z}/{x}/{y}.png'
+    )
     print(f'Updating interferogram: {layer}')
```
Comment (drotheram) [inline app/pages/site.py:490]: An error, print statement ok temporarily, should be displayed if the rastertile does not exist

**Review by drotheram — CHANGES_REQUESTED** [review]
Looks good, some changes needed:

- Remove CoherenceMatrix and bperp CSV files. Add to gitignore if you need to
- Some missing and extra modules imported
- Update sample_config.ini 
- Reference to Geoserver still present in overview.py (line 257)
- error handling for missing rastertile

File: app/Data/Meager/5M10/CoherenceMatrix.csv
[the remark this answers is not in the document]
Comment (mssond) [reply app/Data/Meager/5M10/CoherenceMatrix.csv:1]: Removed all instance of CoherenceMatrix.csv and bperp_all files and then added these two as wildcard in .gitignore

File: app/Data/Meager/5M10/bperp_all
[the remark this answers is not in the document]
Comment (mssond) [reply app/Data/Meager/5M10/bperp_all:1]: Removed all instance of CoherenceMatrix.csv and bperp_all files and then added these two as wildcard in .gitignore

File: app/pages/site.py [hunk 2]
```diff
@@ -42,36 +43,6 @@ def get_config_params(args):
     return config_obj
```
[the remark this answers is not in the document]
Comment (mssond) [reply app/pages/site.py:47]: WMSTileLayer was removed

File: app/pages/site.py [hunk 3]
```diff
@@ -459,39 +461,36 @@ def calc_polygon_centroid(coordinates):
...
+           component_property='url',
            allow_duplicate=True),
     Input(component_id='coherence-matrix', component_property='clickData'),
+    Input('site-dropdown', 'value'),
     prevent_initial_call=True)
-def update_interferogram(click_data):
+def update_interferogram(click_data, target_id):
     """Update interferogram display."""
+    if not target_id:
+        raise PreventUpdate
```
[the remark this answers is not in the document]
Comment (mssond) [reply app/pages/site.py:476]: "from dash.exceptions import PreventUpdate" was added

File: app/Data/Garibaldi/3M23/CoherenceMatrix.csv
Comment (mssond) [reply app/Data/Garibaldi/3M23/CoherenceMatrix.csv:1]: the files are now removed

File: app/pages/site.py (hunk 1 above)
Comment (mssond) [reply app/pages/site.py]: This has been fixed

**Review by drotheram — APPROVED** [review]
Changes look good to me

**Comment by drotheram** [comment]
Create Release of final version of dashboard using geoserver. Possible edge use cases

**Pull request merged**